src/SFML/Graphics/Shader.cpp

```diff
@@ -756,11 +756,11 @@ void Shader::setUniformArray(const std::string& name, const Glsl::Mat3* matrixAr
 ////////////////////////////////////////////////////////////
 void Shader::setUniformArray(const std::string& name, const Glsl::Mat4* matrixArray, std::size_t length)
 {
-    const std::size_t matrixSize = 4 * 4;
+    static const std::size_t matrixSize = matrixArray[0].array.size();
```

Comment:
```suggestion
    static constexpr std::size_t matrixSize = matrixArray[0].array.size();
```

Reply:
`matrixArray[0]` is not a compile time expression